src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -3545,11 +3541,11 @@ void ride_construction_toolupdate_construct(const ScreenCoordsXY& screenCoords)
     {
         for (int zAttempts = 0; zAttempts < numAttempts; ++zAttempts)
         {
+            CoordsXYZ trackPos{};
             window_ride_construction_update_state(
-                &trackType, &trackDirection, &rideIndex, &liftHillAndAlternativeState, &mapCoords->x, &mapCoords->y, &z,
-                nullptr);
+                &trackType, &trackDirection, &rideIndex, &liftHillAndAlternativeState, &trackPos, nullptr);
             _currentTrackPrice = place_provisional_track_piece(
-                rideIndex, trackType, trackDirection, liftHillAndAlternativeState, mapCoords->x, mapCoords->y, z);
+                rideIndex, trackType, trackDirection, liftHillAndAlternativeState, trackPos);
```

Comment:
Is `z` still used in this function? It doesn't get updated anymore after this change, even though it used to.

Reply:
It is, nice catch!